components/Buttons/src/MDCFloatingButton+Animation.m

```diff
@@ -89,110 +86,190 @@ + (float)fab_dragCoefficient {
 }
 #endif
 
++ (CFTimeInterval)fab_animationDurationFromWallTimeDuration:(CFTimeInterval)duration
+                                                   forLayer:(CALayer *)layer {
+  CFTimeInterval finalDuration = duration;
+
+#if TARGET_IPHONE_SIMULATOR
+  finalDuration *= [self fab_dragCoefficient];
+#endif
+
+  return [layer convertTime:finalDuration fromLayer:nil];
+}
+
++ (CFTimeInterval)fab_animationBeginTimeFromWallTimeOffset:(CFTimeInterval)offset
+                                                  forLayer:(CALayer *)layer {
+  CFTimeInterval beginTime = offset;
+
+#if TARGET_IPHONE_SIMULATOR
+  beginTime *= [self fab_dragCoefficient];
+#endif
+
+  beginTime += CACurrentMediaTime();
+  return [layer convertTime:beginTime fromLayer:nil];
+}
+
 - (void)expand:(BOOL)animated completion:(void (^_Nullable)(void))completion {
+  // If both X- and Y-scale values are 1 or greater, do not expand
+  if (self.layer.transform.m11 >= 1 && self.layer.transform.m22 >= 1) {
+    NSAssert(NO, @"The FAB should not be expanded unless its X and Y scales are less than 1.");
```

Comment:
This assertion introduces a runtime breaking change. If we keep it we must document this new behavior and ensure that we bump the major release.

I'd vote for not introducing such a harsh failure case though. If expand is invoked multiple times then a silent failure would likely be sufficient.

Reply:
Done